### ISSUE TITLE
Add Logistic Distribution

### ISSUE THREAD

**Issue by joshdunnlime** [StatMixedML/XGBoostLSS#96]
I would like to add the [Logistic distribution](https://en.wikipedia.org/wiki/Logistic_distribution) for further compatibility with [xgblss](https://skpro.readthedocs.io/en/v2.9.2/api_reference/auto_generated/skpro.regression.xgboostlss.XGBoostLSS.html) [skpro](https://skpro.readthedocs.io/en/v2.9.2/api_reference/auto_generated/skpro.distributions.Logistic.html).

There is no Logistic distribution in pytorch but there are details on how to implement this as a [transformed sigmoid](https://docs.pytorch.org/docs/stable/distributions.html#transformeddistribution). Alternatively, there is an implementation of this in [pyro](https://docs.pyro.ai/en/dev/distributions.html#logistic). Would either of these be preferred?

It should then be trivial to add this to skpro.

Happy to create a PR on this but would be good to see if there is appetite for this and which of the pytorch/pyro distributions should be used.

@fkiraly

**Comment by StatMixedML** [comment]
@joshdunnlime Shouldn't be too difficult, given that it is avaulable in Pyro

https://docs.pyro.ai/en/dev/distributions.html#logistic

**Comment by StatMixedML** [comment]
@joshdunnlime I added the Logistic Distribution.
